File: Easy/C#/NodesDepthSum.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class NodesDepthSum
{
    public class Node
    {
        public int value;
        public Node left;
        public Node right;

        public Node(int value)
        {
            this.value = value;
        }

    }

    public class Level
    {
        public Node root;
        public int height;

        public Level(Node root, int height)
        {
            this.root = root;
            this.height = height;
        }
    }

    public Node root;

    // Return the depth sum of nodes in binary tree
    public int nodeDepthSum(Node node)
    {
        return nodeDepthSum(node, 0);
    }

    // O(n) time | O(l)  n is number of node and l is the height
    public int nodeDepthSum(Node node, int level)
    {
        if (node == null)
        {
            return 0;
        }

        return level + nodeDepthSum(node.left, level + 1) + nodeDepthSum(node.right, level + 1);
    }

    public int nodeDepthSumIterator(Node root){
        int sumOfDepths = 0;
        Stack<Level> stack = new Stack<Level>();
        stack.Push(new Level(root, 0));
        while(stack.Count() > 0){
            Level info = stack.Pop();

            Node node = info.root;
            int height = info.height;
            if(node == null) continue;

            sumOfDepths += height;
            stack.Push(new Level(node.left, height + 1));
            stack.Push(new Level(node.right, height + 1));
        }

        return sumOfDepths;
    }

    private Boolean isEmpty()
    {
        return root == null;
    }

}

File: Easy/JAVASCRIPT/NodesDepthSum.js
function nodesDepthSum(root) {
    if (root === null) return;

    return nodesDepthSumHelper(root, 0);
}

function nodesDepthSumHelper(root, height) {
    if (root === null) return 0;

    return height + nodesDepthSumHelper(root.left, height + 1) + nodesDepthSumHelper(root.right, height + 1);
}

function nodeDepths(root) {
    sumOfDepths = 0;
    const stack = [{ "node": root, "height": 0 }];

    while(stack.length > 0){
        const { node, height } = stack.pop();
        if (node === null) continue;
        sumOfDepths += height
        stack.push({ "node": node.left, "height": height + 1 })
        stack.push({ "node": node.right, "height": height + 1 })
    }

    return sumOfDepths
}


class Node {
    constructor(value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }
}

const root = new Node(10);
root.left = new Node(5);
root.left.left = new Node(2);
root.left.left.left = new Node(1);
root.left.right = new Node(5);
root.right = new Node(15);
root.right.left = new Node(13);
root.right.left.right = new Node(14);
root.right.right = new Node(22);


// const result = nodesDepthSum(root);
// console.log(result);

const result = nodeDepths(root);
console.log(result);